client/landing/stepper/hooks/use-is-site-big-sky-eligible.ts

```diff
@@ -25,7 +25,7 @@ export function useIsBigSkyEligible( flowName?: string ) {
 	const { isOwner } = useIsSiteOwner();
 	const site = useSite();
 	const product_slug = site?.plan?.product_slug || '';
-	const onSupportedDevice = userAgent.isTablet || userAgent.isDesktop;
+	const onSupportedDevice = isTablet() || isDesktop();
```

Comment:
Our life would be easier if `isDesktop` didn't exist: because `ua-parser-js` doesn't support it. For [good reasons](https://github.com/faisalman/ua-parser-js/issues/182#issuecomment-263115448), the UA string never really says that it's coming from a desktop device.

Looking at #95965 and guessing @dereksmart's intent back then, it seems that `! isMobile()` is the right check we want to do.

Reply:
Looking at the reasons for the `isMobile` check in #95965:

> The UX can now handle narrow viewports, but the mobile keyboard still gets in the way.

it seems that we want to decide according the screen size and avoid screens that are too narrow. For that, we have `isMobile` check from `@automattic/viewport`:
```js
import { isMobile } from '@automattic/viewport';
```
This is used at many places in the Calypso codebase, and the check is based on screen width: `<480px` is considered mobile. In this case that's a better fit than a UA check.

Reply:
Updated to use the `isMobile` viewport helper in b8d8dd506f5.

Separately, I wonder if it would be clearer to distinguish the name of these, e.g. `isMobileViewport` and/or `isMobileUserAgent`.

Reply:
> Separately, I wonder if it would be clearer to distinguish the name of these, e.g. `isMobileViewport` and/or `isMobileUserAgent`.

Although I think "mobile viewport" could be a misnomer to begin with, and we should probably refer to viewports as sizes (small, medium, large, etc.).

I see this is handled a bit better with the `useViewportMatch` hook in `@wordpress/compose`.